src/field_theory/galois.lean

```diff
@@ -0,0 +1,243 @@
+  one_mem' := λ _, rfl,
+  mul_mem' := λ _ _ hx hy _, (congr_arg _ (hy _)).trans (hx _),
+  inv_mem' := λ _ hx _, (equiv.symm_apply_eq (alg_equiv.to_equiv _)).mpr (hx _).symm,
+}
+
+
+/-- The fixing_subgroup of `K : intermediate_field F E` is isomorphic to `E ≃ₐ[K] E` -/
+def fixing_subgroup_iso : fixing_subgroup K ≃* (E ≃ₐ[K] E) := {
+  to_fun := λ ϕ, alg_equiv.of_bijective (alg_hom.mk ϕ (alg_equiv.map_one ϕ) (alg_equiv.map_mul ϕ)
+    (alg_equiv.map_zero ϕ) (alg_equiv.map_add ϕ) (ϕ.mem)) (alg_equiv.bijective ϕ),
+  inv_fun := λ ϕ, ⟨alg_equiv.of_bijective (alg_hom.mk ϕ (alg_equiv.map_one ϕ) (alg_equiv.map_mul ϕ)
+    (alg_equiv.map_zero ϕ) (alg_equiv.map_add ϕ) (λ r, ϕ.commutes (algebra_map F K r)))
+      (alg_equiv.bijective ϕ), ϕ.commutes⟩,
+  left_inv := λ _, by {ext, refl},
+  right_inv := λ _, by {ext, refl},
+  map_mul' := λ _ _, by {ext, refl},
+}
+
+theorem fixing_subgroup_of_fixed_field [finite_dimensional F E] :
+  fixing_subgroup (fixed_field H) = H :=
+begin
+  have H_le : H ≤ (fixing_subgroup (fixed_field H)) :=
+    λ ϕ ϕh x, (mul_action.mem_fixed_points E).mp (subtype.mem x) ⟨ϕ, ϕh⟩,
+  suffices : fintype.card H = fintype.card (fixing_subgroup (fixed_field H)),
+  { exact subgroup.ext' (set.eq_of_inclusion_surjective ((fintype.bijective_iff_injective_and_card
+    (set.inclusion H_le)).mpr ⟨set.inclusion_injective H_le, this⟩).2).symm },
+  rw fintype.card_congr (fixing_subgroup_iso (fixed_field H)).to_equiv,
+  rw fintype.card_congr (fixed_points.to_alg_hom_equiv H E),
+  rw fintype.card_congr (algebra_equiv_equiv_algebra_hom (fixed_field H) E),
+  exact fintype.card_congr (by refl),
+end
+
+instance alg_instance : algebra K (fixed_field (fixing_subgroup K)) := {
+  smul := λ x y, ⟨x*y, λ ϕ, by rw [smul_mul', (show ϕ • ↑x = ↑x, by exact subtype.mem ϕ x),
+    (show ϕ • ↑y = ↑y, by exact subtype.mem y ϕ)]⟩,
+  to_fun := λ x, ⟨x, λ ϕ, subtype.mem ϕ x⟩,
+  map_zero' := rfl,
+  map_add' := λ _ _, rfl,
+  map_one' := rfl,
+  map_mul' := λ _ _, rfl,
+  commutes' := λ _ _, mul_comm _ _,
+  smul_def' := λ _ _, rfl,
+}
+
+instance tower_instance : is_scalar_tower K (fixed_field (fixing_subgroup K)) E :=
+⟨λ _ _ _, mul_assoc _ _ _⟩
+
+theorem fixed_field_of_fixing_subgroup [finite_dimensional F E] [h : is_galois F E] :
+  fixed_field (fixing_subgroup K) = K :=
+begin
+  have K_le : K ≤ fixed_field (fixing_subgroup K) := λ x hx ϕ, subtype.mem ϕ (⟨x, hx⟩ : K),
+  suffices : findim K E = findim (fixed_field (fixing_subgroup K)) E,
+  { exact (intermediate_field.eq_of_le_of_findim_eq' K_le this).symm },
+  rw [findim_fixed_field_eq_card, fintype.card_congr (fixing_subgroup_iso K).to_equiv],
+  exact (is_galois_implies_card_aut_eq_findim K E).symm,
+end
+
+lemma card_fixing_subgroup_eq_findim [finite_dimensional F E] [is_galois F E] :
+  fintype.card (fixing_subgroup K) = findim K E :=
+by conv { to_rhs, rw [←fixed_field_of_fixing_subgroup K, findim_fixed_field_eq_card] }
+
+--TODO: Maybe upgrade this to some sort of lattice anti-isomorphism?
```

Comment:
There is actually a notion of `galois_correspondence` in mathlib. I think we can't claim to have done the "Galois correspondence" if we don't show that it is a `galois_correspondence` between `inter_mediate_field F E` and `order_dual (subgroup (E ≃ₐ[F] E))`.

Reply:
There are galois connections and galois insertions. Is this what you're referring to? However, galois connections and galois insertions aren't required to be bijective, I think.

Reply:
also I think the Galois connection in mathlib is monotone, whereas the Galois connection of Galois theory is antitone.

Reply:
Although that would be fixed by the `order_dual`

Reply:
That's true.

Reply:
My bad, I indeed meant `galois_connection`. And there is also `order_iso`, I think. Anyway, the important thing is that we need the theorems about inclusions of subgroups corresponding to (reverse) inclusions of intermediate fields.

Reply:
I added one lemma (`le_iff_le`) and I upgraded the equivalence to an `order_iso`. Is that good enough or should I include some lemmas that follow from the order isomorphism?